packages/providers/src/provider.ts

```diff
@@ -310,9 +310,9 @@ export default class Provider {
   /**
    * Updates the URL for the provider and fetches the consensus parameters for the new URL, if needed.
    */
-  async connect(url: string) {
+  async connect(url: string, options?: ProviderOptions) {
     this.url = url;
-    this.operations = this.createOperations(url);
+    this.operations = this.createOperations(url, options);
     await this.fetchChainAndNodeInfo();
   }
```

Comment:
Should we save the options when we create the instance and in this case just forward the options?

Reply:
I think it doesn't make sense as different networks can have different fetch options

Reply:
In this case, you would create a completely new provider... we can also make it optional and use `options || this.options`.

Reply:
I think saving the options makes sense

Reply:
Yep, and also giving the option to override it on the `connect` method:

```ts
this.operations = this.createOperations(url, options ?? this.options);
```